### PR TITLE
test(smoketest): open services in browser tabs when ready

### PR DESCRIPTION
**Pull request by andrewazores**
Related to #64

To test:

```bash
$ ./smoketest.bash -h
$ ./smoketest.bash -b
$ ./smoketest.bash -Xgtb
```

etc.

### CODE REVIEW

**Review by aali309 — APPROVED** [review]
LGTM

**Comment by mwangggg** [comment]
hmm the tabs aren't opening when I use the `-b` command

**Comment by aali309** [comment]
> hmm the tabs aren't opening when I use the `-b` command

`./smoketest.bash -b` ?

File: smoketest.bash [hunk 1]
```diff
@@ -119,6 +137,40 @@ setupUserHosts() {
...
+        do
+            host="$(echo "${yaml}" | yq ".[${i}].host" | grep -v null)"
+            port="$(echo "${yaml}" | yq ".[${i}].ports[0]" | grep -v null | cut -d: -f1)"
+            if [ -n "${host}" ] && [ -n "${port}" ]; then
+                urls+=("http://${host}:${port}")
+            fi
+        done
+    done
+    set -xe
+    echo "Service URLs:" "${urls[@]}"
```
Comment (mwangggg) [inline smoketest.bash:157]: my logs show the service urls are `http://s3:9001` and `http://cryostat3:8181`, which I believe should be `http://localhost:8181`? Should it also include the links to `grafana` and `db-viewer`?

**Comment by mwangggg** [comment]
> > hmm the tabs aren't opening when I use the `-b` command
> 
> `./smoketest.bash -b` ?

Yes that's the command I'm using and the browser tabs aren't opening.

File: smoketest.bash (hunk 1 above)
Comment (andrewazores) [reply smoketest.bash:157]: `s3:9001` and `cryostat3:8181` are correct, but this assumes that the "userhosts" `LD_PRELOAD` setup is working properly. If that isn't configured then you won't be able to resolve the hostnames `s3` and `cryostat3`, which are supposed to just resolve to `localhost`.

`grafana` and `db-viewer` should be there if you enable those with `smoketest.bash -g` and `smoketest.bash -X` respectively.

**Comment by andrewazores** [comment]
> > > hmm the tabs aren't opening when I use the `-b` command
> > 
> > 
> > `./smoketest.bash -b` ?
> 
> Yes that's the command I'm using and the browser tabs aren't opening.

If you do `xdg-open https://github.com` in your terminal, does that work?

File: smoketest.bash (hunk 1 above)
Comment (andrewazores) [reply smoketest.bash:157]:
```bash
$ cat ~/.hosts 
localhost db
localhost db-viewer
localhost grafana
localhost jfr-datasource
localhost vertx-fib-demo-1
localhost quarkus-test-agent
localhost s3
localhost cryostat3
```

**Comment by mwangggg** [comment]
> > > > hmm the tabs aren't opening when I use the `-b` command
> > > 
> > > 
> > > `./smoketest.bash -b` ?
> > 
> > 
> > Yes that's the command I'm using and the browser tabs aren't opening.
> 
> If you do `xdg-open https://github.com` in your terminal, does that work?
> 
> I think this will still end up being contingent on the userhosts and `LD_PRELOAD` working, since what's happening is that extra bash subshells are forked which poll the services' expected mapped ports and once they come up invokes `xdg-open` on that URL. But the hostname that the subshells are testing with `curl` are the service hostnames like `s3` and `cryostat3`, not `localhost`, so if the userhosts setup isn't working then `curl` will always fail the poll check and never allow the `xdg-open` to run.

yes, that command does open github in my browser

**Comment by andrewazores** [comment]
Okay, so then I think the culprit must be the userhosts thing.

**Comment by andrewazores** [comment]
Try `USE_USERHOSTS=false ./smoketest.bash -b`. This should open things with `localhost` directly instead of trying to use the userhosts/LD_PRELOAD setup. If that works for you then you can add that line to your `~/.bashrc` or similar shell configuration so that it always does that without having to write it out every time.

You might still want to try to fix your userhosts setup at some point because IIRC the way that I set up the jfr-datasource and Grafana dashboard will require it in this smoketest. If you don't fix it then "View in Grafana" won't work, even if everything else does via `localhost`.

**Review by mwangggg — APPROVED** [review]
looks good to me

**Pull request merged**